### PR TITLE
ERM-291: Add usage data provider to subscription agreement

### PR DESCRIPTION
**Pull request by CalamityC**
change SubscriptionAgreement domain model
add linked-udp.groovy to database changelog

### CODE REVIEW

**Comment by doytch** [comment]
Could we call them something like `usageDataProviderId` and `usageDataProviderNote` so it's a bit more obvious what they are when looking at this two years from now? 😆 

Side-note, should the fetch for the actual provider record be done on the backend or frontend? @sosguthorpe?

**Comment by CalamityC** [comment]
> Given it's only 1 record we should probably fetch it and embed it. I added a comment to the issue in JIRA regarding how we annotate the ID property to do that. Take a look.

If we want to do that, we would need mod-erm-usage, right? Currently it's not part of the vagrant box.

**Comment by sosguthorpe** [comment]
> If we want to do that, we would need mod-erm-usage, right? Currently it's not part of the vagrant box.
Yes you would.

However, it has been confirmed by Owen that we do need multiple (0-N) usage-data providers so the properties here are not enough and you will need a new domain object to represent the link. 

Take a look at the [RemoteLicenceLink](https://github.com/folio-org/mod-agreements/blob/master/service/grails-app/domain/org/olf/erm/RemoteLicenseLink.groovy) as an example of a full object that has a link. The key here is extending the RemoteOkapiLink class. You'll just need to rename it and remove the status.

File: service/grails-app/domain/org/olf/erm/UsageDataProvider.groovy [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+	}
+  
+	static constraints = {
+		owner(nullable:false, blank:false)
+        usageDataProviderNote(nullable:true, blank:false)
+	}
+
+	@Override
+	public String remoteUri() {
+		return 'eusage/usage-data-providers';
```
Comment (sosguthorpe) [inline service/grails-app/domain/org/olf/erm/UsageDataProvider.groovy]: The path for the usage data provider info is just `usage-data-providers` not `eusage/usage-data-providers`

File: service/grails-app/domain/org/olf/erm/SubscriptionAgreement.groovy [hunk 2]
```diff
@@ -73,15 +73,17 @@ public class SubscriptionAgreement implements MultiTenant<SubscriptionAgreement>
...
   ]
 
   static mappedBy = [
     items: 'owner',
     historyLines: 'owner',
     contacts: 'owner',
     orgs: 'owner',
-    linkedLicenses: 'owner'
+    linkedLicenses: 'owner',
+	usageDataProviders: 'owner',
```
Comment (sosguthorpe) [inline service/grails-app/domain/org/olf/erm/SubscriptionAgreement.groovy]: Indentation

File: service/grails-app/domain/org/olf/erm/SubscriptionAgreement.groovy [hunk 3]
```diff
@@ -134,7 +137,7 @@ public class SubscriptionAgreement implements MultiTenant<SubscriptionAgreement>
              description(nullable:true, blank:false)
                   vendor(nullable:true, blank:false)
        attachedLicenceId(nullable:true, blank:false)
-	   		     licenseNote(nullable:true, blank:false)
+	   		 licenseNote(nullable:true, blank:false)
```
Comment (sosguthorpe) [inline service/grails-app/domain/org/olf/erm/SubscriptionAgreement.groovy]: Indentation

File: service/grails-app/domain/org/olf/erm/SubscriptionAgreement.groovy [hunk 4]
```diff
@@ -73,15 +73,17 @@ public class SubscriptionAgreement implements MultiTenant<SubscriptionAgreement>
     externalLicenseDocs: DocumentAttachment,
                    docs: DocumentAttachment,
          linkedLicenses: RemoteLicenseLink,
-      supplementaryDocs: DocumentAttachment
+      supplementaryDocs: DocumentAttachment,
+	 usageDataProviders: UsageDataProvider,
```
Comment (sosguthorpe) [inline service/grails-app/domain/org/olf/erm/SubscriptionAgreement.groovy]: Indentation

File: service/grails-app/controllers/org/olf/UrlMappings.groovy [hunk 5]
```diff
@@ -31,6 +31,19 @@ class UrlMappings {
           method = 'GET'
         }
       }
+	  
+	  '/usageDataProviders' {
```
Comment (sosguthorpe) [inline service/grails-app/controllers/org/olf/UrlMappings.groovy]: Indentation

File: service/grails-app/controllers/org/olf/UrlMappings.groovy [hunk 6]
```diff
@@ -31,6 +31,19 @@ class UrlMappings {
...
+		  controller = 'usageDataProvider'
+		  method = 'GET'
+		  filters = { "owner==${params.subscriptionAgreementId}" }
+	  }
+	  
+	  collection {
```
Comment (sosguthorpe) [inline service/grails-app/controllers/org/olf/UrlMappings.groovy]: Indentation

**Pull request merged**